addons/l10n_eg_edi_eta/models/account_move.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from datetime import datetime
+
+from odoo.exceptions import ValidationError, UserError
+
+from odoo import models, fields, api, _
+import logging
+
+import json
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_eg_submission_id = fields.Char(string='Submission ID', compute='compute_eta_response_data', store=True, copy=False)
+    l10n_eg_uuid = fields.Char(string='Document UUID', compute='compute_eta_response_data', store=True, copy=False)
+    l10n_eg_eta_invoice_id = fields.Many2one('ir.attachment', copy=False)
+    l10n_eg_signing_time = fields.Datetime('Signing TIme', copy=False)
+    l10n_eg_is_signed = fields.Boolean(copy=False)
+
+    @api.depends('l10n_eg_eta_invoice_id.raw')
+    def compute_eta_response_data(self):
+        for rec in self:
+            response_data = rec.l10n_eg_eta_invoice_id and json.loads(rec.l10n_eg_eta_invoice_id.raw).get('response')
+            if response_data:
+                rec.l10n_eg_uuid = response_data.get('l10n_eg_uuid')
+                rec.l10n_eg_submission_id = response_data.get('l10n_eg_submission_id')
+            else:
+                rec.l10n_eg_uuid = False
+                rec.l10n_eg_submission_id = False
+
+    def button_draft(self):
+        self.l10n_eg_eta_invoice_id.unlink()
+        return super().button_draft()
+
+    def action_post_sign_invoice(self):
+        company_ids = self.mapped('company_id')
+        # since the middleware accepts only one drive at a time, we have to limit signing to one company at a time
+        if len(company_ids) > 1:
+            raise UserError(_('Please only sign invoices from once company at a time'))
+
+        company_id = company_ids[0]
+        drive_id = self.env['l10n_eg_edi.thumb.drive'].search([('user_id', '=', self.env.user.id),
+                                                               ('company_id', '=', company_id.id)])
+
+        if not drive_id:
+            raise ValidationError(_('Please setup a drive for company %s', company_id.name))
+
+        if not drive_id.certificate:
+            raise ValidationError(_('Please setup the certificate on the thumb drive menu'))
+
+        self.write({
+            'l10n_eg_signing_time': datetime.now()
```

Comment:
signing time is UTC?

Reply:
all server-side timestamps are supposed to be stored in UTC, yes